File: python/LeetCode/02452.py
from typing import List

def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
    query_length = len(queries)
    dict_legnth = len(dictionary)
    word_length = len(queries[0])
    output = []

    for i in range(query_length):
        for j in range(dict_legnth):
            edits = 0
            for k in range(word_length):
                if queries[i][k] != dictionary[j][k]:
                    edits += 1

            if edits <= 2:
                output.append(queries[i])
                break

    return output

File: go/leetcode/02452.go
package main

func twoEditWords(queries []string, dictionary []string) []string {
	wordLength := len(queries[0])
	output := make([]string, 0, len(queries))

	edits := 0
	for _, word := range queries {
		for _, candidate := range dictionary {
			edits = 0
			for i := range wordLength {
				if word[i] != candidate[i] {
					edits += 1
				}
				if edits > 2 {
					break
				}
			}
			if edits <= 2 {
				output = append(output, word)
				break
			}
		}
	}

	return output
}
